newIDE/app/src/ProjectsStorage/ResourceMover/index.js

```diff
@@ -0,0 +1,276 @@
+// @flow
+import { Trans } from '@lingui/macro';
+import * as React from 'react';
+import Dialog, { DialogPrimaryButton } from '../../UI/Dialog';
+import FlatButton from '../../UI/FlatButton';
+import { Line } from '../../UI/Grid';
+import Text from '../../UI/Text';
+import {
+  Table,
+  TableBody,
+  TableHeader,
+  TableHeaderColumn,
+  TableRow,
+  TableRowColumn,
+} from '../../UI/Table';
+import { ColumnStackLayout } from '../../UI/Layout';
+import LinearProgress from '../../UI/LinearProgress';
+import { type StorageProviderOperations, type StorageProvider } from '../index';
+import { type AuthenticatedUser } from '../../Profile/AuthenticatedUserContext';
+import { type FileMetadata } from '..';
+
+export type MoveAllProjectResourcesOptionsWithoutProgress = {|
+  project: gdProject,
+  oldFileMetadata: FileMetadata,
+  oldStorageProvider: StorageProvider,
+  oldStorageProviderOperations: StorageProviderOperations,
+  newFileMetadata: FileMetadata,
+  newStorageProvider: StorageProvider,
+  newStorageProviderOperations: StorageProviderOperations,
+  authenticatedUser: AuthenticatedUser,
+|};
+
+export type MoveAllProjectResourcesOptions = {|
+  ...MoveAllProjectResourcesOptionsWithoutProgress,
+  onProgress: (number, number) => void,
+|};
+
+export type MoveAllProjectResourcesResult = {|
+  erroredResources: Array<{|
+    resourceName: string,
+    error: Error,
+  |}>,
+|};
+
+export type MoveAllProjectResourcesFunction = (
+  options: MoveAllProjectResourcesOptions
+) => Promise<MoveAllProjectResourcesResult>;
+
+export type ResourceMover = {|
+  moveAllProjectResources: MoveAllProjectResourcesFunction,
+|};
+
+type ResourceMoverDialogProps = {|
+  progress: number,
+  result: ?MoveAllProjectResourcesResult,
+  onAbandon: ?() => void,
+  onRetry: ?() => void,
+  genericError: ?Error,
+|};
+
+const styles = {
+  tableCell: {
+    // Avoid long filenames breaking the design.
+    wordBreak: 'break-word',
+  },
+};
+
+export const ResourceMoverDialog = ({
+  progress,
+  result,
+  onAbandon,
+  onRetry,
+  genericError,
+}: ResourceMoverDialogProps) => {
+  const hasErrors =
+    (result && result.erroredResources.length > 0) || !!genericError;
+
+  return (
+    <Dialog
+      actions={[
+        onRetry ? (
+          <DialogPrimaryButton
```

Comment:
I think the retry button should be on the right side.
One might be tempted to ignore without reading the dialog:
![image](https://user-images.githubusercontent.com/32449369/186874358-fa15c97a-e9a1-484c-b602-65ad25bab5ed.png)

Reply:
Fixed :) And factored for usage in other places.